src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java

```diff
@@ -248,14 +252,22 @@ private void getInstanceList() {
 
                     if (parts.length >= 2) {
                         String nodeType = parts[1];
+                        String nodeName = parts[0];
+                        String nodeInstallationPath = parts[3];
```

Comment:
You are checking if `parts.length >= 2`, but later accessing `parts[3]`. This could cause an `ArrayIndexOutOfBoundsException` if the length of parts is less than 4.